config/routes.rb

```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
-  get  'v1/:site_name',                     to: 'api/version1#get_proxy', defaults: { older_than: -1 }
-  post 'v1/:site_name/:proxy_id/succeeded', to: 'api/version1#report_result', succeeded: true
-  post 'v1/:site_name/:proxy_id/failed',    to: 'api/version1#report_result', succeeded: false
+  get  'v1/:site_name',                     to: 'api/version1#get_proxy', constraints: { site_name: %r{[a-z0-9_.-]+} }
```

Comment:
It would be nice if these lines weren't nearly so long, but I can appreciate the readability trade off of making these all multiple lines.  Either way, the regex should be documented to describe what it's doing.